aten/src/ATen/core/ivalue.h

```diff
@@ -518,16 +518,16 @@ struct CAFFE2_API IValue final {
   }
 
   // Detection Aliased tensors.
-  struct HashIValue {
+  struct HashAliasedIValue {
     size_t operator()(const IValue& val) const {
       if (val.isTensor()) {
-        return 0;
+        return reinterpret_cast<size_t>(val.toTensor().storage().unsafeGetStorageImpl());
       }
       return val.hash();
     }
   };
```

Comment:
Could you rename `IValue::hash`. Maybe something like : `IValue::unsafeGetPointer` ? and make it private if it doesn't have any external uses.

Reply:
You can remove `hash` and just use `internalToPointer`